controllers/ptpoperatorconfig_controller.go

```diff
@@ -292,44 +301,90 @@ func (r *PtpOperatorConfigReconciler) getAmqStatus(svcName string, namespace str
 		return false
 	}
 	for _, p := range podList.Items {
-		if p.Spec.ServiceAccountName == svcName && p.Status.Phase == "Running" {
-			return true
+		for _, c := range p.Spec.Containers {
+			if c.Name == svcName && p.Status.Phase == "Running" {
+				return true
+			}
 		}
+
 	}
 	return false
 }
 
-func (r *PtpOperatorConfigReconciler) checkAmqStatus(data *render.RenderData, amqSvc string, amqNamespace string) {
-	var timeout time.Duration
-	if r.IsInitialSync {
-		timeout = cancelTimeout
-		r.IsInitialSync = false
-	} else {
-		// no need to retry if it is not initial sync
-		timeout = retryTimeout * 2
+// EventTransportHostAvailabilityCheck ... check availability for transporthost (for amq only )
+func (r *PtpOperatorConfigReconciler) EventTransportHostAvailabilityCheck(transportHost string) (string, error) {
+	var transportUrl *url.URL
+	if transportHost == "" {
+		glog.Warningf("ptp operator config Spec, ptpEventConfig.transportHost=%v is not valid, proceed as %s",
+			transportHost, AmqDefaultHost)
+		return AmqDefaultHost, nil
 	}
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-
-	defer cancel()
-	isFirstFail := true
-	for {
-		select {
-		case <-ctx.Done():
-			glog.Infof("time out checking for AMQP service")
-			data.Data["EventTransportHost"] = "amqp://nohup"
-			return
-		default:
-		}
-		if !r.getAmqStatus(amqSvc, amqNamespace) {
-			if isFirstFail {
-				glog.Infof("AMQP service %v in namespace %v is NOT running, retry every %s for %s",
-					amqSvc, amqNamespace, retryTimeout, timeout)
-				isFirstFail = false
-			}
-			time.Sleep(retryTimeout)
-			continue
+
+	// if new transport has been applied reset everything
+	r.TransportHostStatus.ResetOnChange(transportHost)
+
+	// example transportHost: "amqp://amq-router.amq-router.svc.cluster.local"
+	// if transportHost is "amqp://nohup", ignore any validation and print events to log
+	transportUrl, err := url.Parse(transportHost)
+	if err != nil {
+		glog.Warningf("ptp operator config Spec, ptpEventConfig.transportHost=%v is not valid, proceed as %s",
+			transportHost, AmqDefaultHost)
+		return AmqDefaultHost, nil
+	} else if r.TransportHostStatus.RetryThisHost(transportUrl.Scheme, transportUrl.Host) { // not set to nohup and last try was not success
+		if r.TransportHostStatus.RetryOnCount() {
+			if amq := strings.Split(transportUrl.Host, "."); len(amq) > 1 { // check for availability if its amq
+				glog.Infof("check AMQP service %v in namespace %v before deploying linuxptp daemon (%d of %d attempts)", amq[0], amq[1],
+					r.TransportHostStatus.TransportHostRetryCount+1, TransportRetryMaxCount)
+				if !r.getAmqStatus(amq[0], amq[1]) {
+					r.TransportHostStatus.Inc()
+					return transportHost, fmt.Errorf("%s", AmqReadyStateError) // failed then try again
+				}
+			} // else continue as it is
```

Comment:
else (if len(amq) == 1) print out the transportUrl.Host and continue as nohup.

Reply:
no we don't it is to support for localhost:5672

Reply:
ok